src/helpers/asHTML.ts

```diff
@@ -203,41 +203,86 @@ const wrapMapSerializerWithStringChildren = (
 	return wrapMapSerializer(modifiedMapSerializer);
 };
 
+/**
+ * Configuration that determines the output of `asHTML()`.
+ */
+type AsHTMLConfig = {
+	/**
+	 * An optional link resolver function to resolve links.
+	 * Without it you're expected to use the `routes` options from the API.
+	 */
+	linkResolver?: LinkResolverFunction<string | null | undefined> | null;
+
+	/**
+	 * An optional Rich Text Serializer, unhandled cases will fallback to the default serializer
+	 */
+	htmlRichTextSerializer?: HTMLRichTextSerializer | null;
+};
+
+// TODO: Remove when we remove support for deprecated tuple-style configuration.
+/**
+ * @deprecated Use object-style configuration instead.
+ */
+type AsHTMLDeprecatedTupleConfig = [
+	linkResolver?: LinkResolverFunction<string | null | undefined> | null,
+	htmlRichTextSerializer?: HTMLRichTextSerializer | null,
+];
+
 /**
  * The return type of `asHTML()`.
  */
 type AsHTMLReturnType<Field extends RichTextField | null | undefined> =
 	Field extends RichTextField ? string : null;
 
 /**
- * Serializes a Rich Text or Title field to an HTML string
+ * Serializes a Rich Text or Title field to an HTML string.
  *
  * @param richTextField - A Rich Text or Title field from Prismic
- * @param linkResolver - An optional link resolver function to resolve links,
- *   without it you're expected to use the `routes` options from the API
- * @param htmlRichTextSerializer - An optional Rich Text Serializer, unhandled cases will fallback
- *   to the default serializer
+ * @param configObjectOrTuple - Configuration that determines the output of `asHTML()`
  *
  * @returns HTML equivalent of the provided Rich Text or Title field
  * @see Templating Rich Text and title fields from Prismic {@link https://prismic.io/docs/template-content-vanilla-javascript#rich-text-and-title}
  */
 export const asHTML = <Field extends RichTextField | null | undefined>(
 	richTextField: Field,
-	linkResolver?: LinkResolverFunction<string | null | undefined> | null,
-	htmlRichTextSerializer?: HTMLRichTextSerializer | null,
+	// TODO: Rename to `config` when we remove support for deprecated tuple-style configuration.
+	...configObjectOrTuple: [config?: AsHTMLConfig] | AsHTMLDeprecatedTupleConfig
 ): AsHTMLReturnType<Field> => {
 	if (richTextField) {
+		// TODO: Remove when we remove support for deprecated tuple-style configuration.
+		const [configObjectOrLinkResolver, maybeHTMLRichTextSerializer] =
+			configObjectOrTuple;
+		let config: AsHTMLConfig;
+		if (
+			typeof configObjectOrLinkResolver === "function" ||
+			configObjectOrLinkResolver == null
+		) {
+			config = {
+				linkResolver: configObjectOrLinkResolver,
+				htmlRichTextSerializer: maybeHTMLRichTextSerializer,
+			};
+		} else {
+			config = { ...configObjectOrLinkResolver };
+		}
```

Comment:
This seems a bit complex. Is there any way to simplify this into a single `config` reassignment?

```ts
config = {
  linkResolver: ...,
  htmlRichTextSerializer: ...,
}
```

Reply:
I could refactor to a ternary, but I don't see the benefit in readability tbh 🤔

Reply:
No worries! If you don't see a benefit or better way of doing it, let's keep it as is.